client/src/components/browse-recipes/Category.svelte

```diff
@@ -0,0 +1,104 @@
+<script>
+  import { createEventDispatcher, onMount } from 'svelte';
+  import categoriesApi from '@/api/categoriesApi';
+  const dispatch = createEventDispatcher();
+  addEventListener('DOMContentLoaded', event => {
+    let element = document.getElementById('categorySelect');
+
+    element.addEventListener('change', () => {
+      dispatch('categoryToggle', {
+        // @ts-ignore
+        value: element.value,
+        // @ts-ignore
+        text: element.options[element.selectedIndex].text,
+      });
+    });
+  });
+  let categories = [];
+  onMount(async () => {
+    expanded = false;
+    categories = await categoriesApi.categories();
+  });
+
+  let expanded;
+  function expand() {
+    let element = document.querySelector('.categoryContainer');
+    if (expanded == false) {
+      // @ts-ignore
+      element.style.visibility = 'visible';
+      // @ts-ignore
+      element.style.height = '10vh';
+      expanded = true;
+    } else {
+      // @ts-ignore
+      element.style.visibility = 'hidden';
+      // @ts-ignore
+      element.style.height = '0vh';
+      expanded = false;
+    }
+  }
+</script>
+
+<main>
+  <!-- svelte-ignore a11y-click-events-have-key-events -->
+  <div class="expand" on:click={expand}>Filteri</div>
```

Comment:
we shouldn't ignore this error cause it actually makes sense. This can be fixed if you just change this to a button instead of div